src/main/java/net/mcreator/element/parts/procedure/RetvalProcedure.java

```diff
@@ -29,6 +29,7 @@
 	public static final Map<Class<? extends RetvalProcedure<?>>, JsonDeserializer<? extends RetvalProcedure<?>>> GSON_ADAPTERS = new HashMap<>() {{
 		put(NumberProcedure.class, new NumberProcedure.GSONAdapter());
 		put(StringProcedure.class, new StringProcedure.GSONAdapter());
+		put(LogicProcedure.class, new LogicProcedure.GSONAdapter());

```

Comment:
Maybe put this before number procedure to keep the `logic, number, text` order we have practically everywhere so far?

Reply:
It doesn't affect things in UI. Don't worry